_sources/SequenceMutation/ObjectsandReferences.rst

```diff
@@ -0,0 +1,88 @@
+..  Copyright (C)  Brad Miller, David Ranum, Jeffrey Elkner, Peter Wentworth, Allen B. Downey, Chris
+    Meyers, and Dario Mitchell.  Permission is granted to copy, distribute
+    and/or modify this document under the terms of the GNU Free Documentation
+    License, Version 1.3 or any later version published by the Free Software
+    Foundation; with Invariant Sections being Forward, Prefaces, and
+    Contributor List, no Front-Cover Texts, and no Back-Cover Texts.  A copy of
+    the license is included in the section entitled "GNU Free Documentation
+    License".
+
+.. qnum::
+   :prefix: seqmut-3-
+   :start: 1
+
+Objects and References
+----------------------
+
+If we execute these assignment statements,
+
+.. sourcecode:: python
+    
+    a = "banana"
+    b = "banana"
+
+we know that ``a`` and ``b`` will refer to a string with the letters
+``"banana"``. But we don't know yet whether they point to the *same* string.
+
+There are two possible ways the Python interpreter could arrange its internal states:
+
+.. image:: Figures/refdiag1.png
+   :alt: List illustration 
+
+or
+
+.. image:: Figures/refdiag2.png
+   :alt: List illustration
+
+In one case, ``a`` and ``b`` refer to two different string objects that have the same
+value. In the second case, they refer to the same object. Remember that an object is something a variable can
+refer to.
+
+We can test whether two names refer to the same object using the *is*
+operator.  The *is* operator will return true if the two references are to the same object.  In other words, the references are the same.  Try our example from above.
+
+.. activecode:: ac8_3_1
+
+    a = "banana"
+    b = "banana"
+
+    print(a is b)
+
+The answer is ``True``.  This tells us that both ``a`` and ``b`` refer to the same object, and that it
```

Comment:
I think we can add a sentence here to say that 

Python assigns every object a unique id and when we ask ``a is b`` what python is really doing is checking to see if the id(a) == id(b). 

In the example we could even print the id(a) and the id(b)